Source/Core/DolphinQt/Host.cpp

```diff
@@ -85,6 +88,44 @@ void Host::SetMainWindowHandle(void* handle)
   m_main_window_handle = handle;
 }
 
+static void RunWithGPUThreadInactive(std::function<void()> f)
```

Comment:
I believe this could be a template to avoid the overhead of the std function:

```
template <typename Func>
static void RunWithGPUThreadInactive(Func&& f)
{
  if (Core::IsGPUThread())
  {
    f();
  }
  else if (Core::IsCPUThread())
  {
    const bool was_running = Core::GetState() == Core::State::Running;
    Fifo::PauseAndLock(true, was_running);
    f();
    Fifo::PauseAndLock(false, was_running);
}
else
{
  Core::RunAsCPUThread(std::forward(f));
}
}
```

Reply:
The most common case is the `RunAsCPUThread` case, in which `f` has to be turned into an `std::function` anyway. Would this be beneficial despite that?

Reply:
Ah I missed that.  Probably not then.  We may want to consider that in the future (if it's even possible)